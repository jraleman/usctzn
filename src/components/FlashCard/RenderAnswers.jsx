import React from 'react';
import uniqueId from '../../utils/helpers/uniqueId';

const RenderAnswers = ({ children }) => (
    <>
        {children?.map((a) => (
            <li key={uniqueId('answer_')}>{a}</li>
        ))}
    </>
);

export default RenderAnswers;
