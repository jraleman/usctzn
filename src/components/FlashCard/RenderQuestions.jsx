import React from 'react';
import uniqueId from '../../utils/helpers/uniqueId';

const RenderQuestions = ({ children }) => (
    <>
        {children[0]?.map((a) => (
            <li key={uniqueId('answer_')}>{a}</li>
        ))}
    </>
);

export default RenderQuestions;
