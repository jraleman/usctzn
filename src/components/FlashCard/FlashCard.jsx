import React from 'react';
import RenderQuestions from './RenderQuestions';

const FlashCard = ({ question, answers }) => (
    <div>
        <h3>{question}</h3>
        <ol>
            <RenderQuestions>{answers}</RenderQuestions>
        </ol>
    </div>
);

export default FlashCard;
