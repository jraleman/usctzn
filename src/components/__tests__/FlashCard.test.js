import React from 'react';
import { render, screen } from '@testing-library/react';
import FlashCard from '../FlashCard';
import americanGovermentData from '../../data/american-goverment.json';

describe('FlashCard', () => {
    it('renders', async () => {
        const question = '';
        const answers = [''];
        const component = render(<FlashCard question={question} answers={answers} />);
        expect(component).not.toBeNull();
        expect(component).toMatchSnapshot();
    });

    it('shows some answers', async () => {
        const { content } = americanGovermentData;
        const question = Object.keys(content[0])
        const answers = Object.values(content[0])
        const component = render(<FlashCard question={question} answers={answers} />);
        const title = screen.getByText(question)?.textContent;
        expect(title).toEqual(question[0]);
        expect(answers[0]).toEqual(
            expect.arrayContaining([expect.any(String)])
        );
        expect(component).not.toBeNull();
        expect(component).toMatchSnapshot();
    });
});
