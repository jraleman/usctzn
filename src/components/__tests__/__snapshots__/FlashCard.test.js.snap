// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashCard renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h3 />
        <ol>
          <li />
        </ol>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h3 />
      <ol>
        <li />
      </ol>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FlashCard shows some answers 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h3>
          What is the form of government of the United States?
        </h3>
        <ol>
          <li>
            Republic
            Constitution-based federal republic
            Representative democracy
          </li>
        </ol>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h3>
        What is the form of government of the United States?
      </h3>
      <ol>
        <li>
          Republic
          Constitution-based federal republic
          Representative democracy
        </li>
      </ol>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
